### PR TITLE
add next auth

### PR DESCRIPTION
**Pull request by NCain24**
## Summary of Work

* Add authorization with Next-Auth.js.

Ticket  https://github.com/NCain24/restaurant-app/issues/8

### CODE REVIEW

**Comment by jared-cain** [comment]
It looks like the deployment is failing from vercel. Can you see what the error is?

File: src/app/search/page.tsx [hunk 1]
```diff
@@ -1,46 +1,55 @@
...
+
+const SearchPage = () => {
+  const { data: session, status } = useSession();
+
+  const [searchTerm, setSearchTerm] = useState('');
+  const [filteredRestaurants, setFilteredRestaurants] = useState([]);
+
+  if (status === 'unauthenticated') {
+    redirect('/');
+    // return <p className='pt-20 text-xl'>Please sign in or sign up</p>
```
Comment (jared-cain) [inline src/app/search/page.tsx]: remove comment

File: src/app/layout.tsx [hunk 2]
```diff
@@ -1,23 +1,25 @@
...
+import Header from './components/Header';
+import Providers from './providers';
+import './globals.css';
+
+export const metadata: Metadata = {
+  title: 'Create Next App',
+  description: 'Generated by create next app',
+};
+
+export default async function RootLayout({
```
Comment (jared-cain) [inline src/app/layout.tsx]: Does this need to be async? If not can we remove?

File: src/app/components/HeaderAuth.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
+'use client';
+
+import { useSession } from 'next-auth/react';
+import * as actions from '@/actions';
+
+export default function HeaderAuth() {
+  const session = useSession();
+
+  let authContent: React.ReactNode;
```
Comment (jared-cain) [inline src/app/components/HeaderAuth.tsx]: You can refactor this whole block to remove `authContent` and just return null or the elements when the condition is met

**Review by jared-cain — COMMENTED** [review]
Left a few comments but looking good!

**Review by jared-cain — APPROVED** [review]
In the future we shouldnt use beta versions of packages like we have with next-auth. Other than that, LGTM.

**Comment by NCain24** [comment]
> In the future we shouldnt use beta versions of packages like we have with next-auth. Other than that, LGTM.

I will ask you about best practices on choosing the correct packages in the future if I have any questions!

**Pull request merged**